src/controller/petitions.py

```diff
@@ -1,65 +1,105 @@
 
-def new_petition(data: Petition.Create):
-    db_petition = Petitions(
-        title=data.title,
-        contents=data.contents,
-        proposal=data.proposal,
-        petitioner=str(data.petitioner),
-    )
-    con = db.session
-    con.add(db_petition)
-    con.commit()
-    con.refresh(db_petition)
-    return {"id": db_petition.petition_id}
+class PetitionController:
+    def __init__(
+        self,
+        id: int = None,
+        status: int = None,
+        petitioner: int = None,
+        title: str = None,
+        contents: str = None,
+        proposal: str = None,
+        agreed: int = None,
+        created_at: datetime = None,
+        end_at: datetime = None,
+        agreeable: bool = None,
+        answer: str = None,
+        answered_at: datetime = None,
+        answered_by: str = None,
+    ):
+        self.id = id
+        self.status = status
+        self.petitioner = petitioner
+        self.title = title
+        self.contents = contents
+        self.proposal = proposal
+        self.agreed = agreed
+        self.created_at = created_at
+        self.end_at = end_at
+        self.agreeable = agreeable
+        self.answer = answer
+        self.answered_at = answered_at
+        self.answered_by = answered_by
 
+    def create(self):
+        db_petition = Petitions(
+            title=self.title,
+            contents=self.contents,
+            proposal=self.proposal,
+            petitioner=str(self.petitioner),
+        )
+        con = db.session
+        con.add(db_petition)
+        con.commit()
+        con.refresh(db_petition)
+        return {"id": db_petition.petition_id}
+
+    def consent(self):
+        con = db.session
 
-def counting_petition():
-    con = db.session
+        result = con.query(Agreements).filter(Agreements.petition_id == self.id).first()
 
-    total = con.query(Petitions).count()
-    answered = con.query(Petitions).filter(Petitions.status == "answered").count()
-    pending = con.query(Petitions).filter(Petitions.status == "pending").count()
+        if result:
+            # 유저 아이디또한 존재하면 return 400
+            # 없다면 return 200
+            return 400
+        return 404
 
-    return {"total": total, "answered": answered, "pending": pending}
+    def load(self):
+        return
 
+    def delete(self):
+        return
 
-def consent_petition(id: int):
-    con = db.session
+    @staticmethod
+    def count_petitions():
+        con = db.session
 
-    result = con.query(Agreements).filter(Agreements.petition_id == id).first()
+        total = con.query(Petitions).count()
+        answered = con.query(Petitions).filter(Petitions.status == "answered")
+        pending = con.query(Petitions).filter(Petitions.status == "pending").count()
 
-    if result:
-        # 유저 아이디또한 존재하면 return 400
-        # 없다면 return 200
-        return 400
-    return 404
+        return {"total": total, "answered": answered, "pending": pending}
 
+    @staticmethod
+    def search_petitions():
```

Comment:
```suggestion
    def search_petitions(q: str, page: int):
```
검색할 때 인자값이 필요하니 작성하는것이 나아보여요 'ㅅ'